### PR TITLE
create screen casting and reverse tethering doc

### PR DESCRIPTION
**Pull request by lduboeuf**
don't know if this is the right place for this memo

### CODE REVIEW

File: userguide/advanceduse/reverse-tethering.rst [hunk 1]
```diff
@@ -0,0 +1,32 @@
+Reverse tethering
+=================
+
+Some users may don't have an available wifi connection for their phone nor a data subscription from their mobile provider.
+This short tutorial will help you to connect your Ubuntu Touch to your computer to access internet.
+
+Steps:
+------
+
+1 - on phone: ``android-gadget-service enable rndis``
```
Comment (UniversalSuperBox) [inline userguide/advanceduse/reverse-tethering.rst]: This numbering style does not work in the build output. Please use the following:

1. 

2.

File: userguide/advanceduse/screencasting.rst [hunk 2]
```diff
@@ -0,0 +1,34 @@
+Screen Casting your UT device to your computer
+==============================================
+
+
+Ubuntu Touch comes with a command line utility called ``mirscreencast`` which dumps screen frames to a file.
+The idea here is to stream UT display to a listening computer over the network so that we can either watch it live or record it to a file.
+
+On computer side:
+-----------------
+  For real time casting:
```
Comment (UniversalSuperBox) [inline userguide/advanceduse/screencasting.rst]: Please do not indent here.

File: userguide/advanceduse/screencasting.rst [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+  Prepare your computer to listen to a tcp port(here 1234) and forward stream to a video player (here mplayer):
+
+      nc -l -p 1234 | gzip -dc | mplayer -demuxer rawvideo -rawvideo w=384:h=640:format=rgba -
```
Comment (UniversalSuperBox) [inline userguide/advanceduse/screencasting.rst]: Use two colons followed by an indented line to create a code block. For example...

```
[...]and forward stream to a video player (here mplayer)::

    nc -l[...]
```

File: userguide/advanceduse/screencasting.rst [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  
+
+  Prepare your computer to listen to a tcp port(here 1234) and forward raw stream to a video encoder (here mencoder):
+
+      nc -l -p 1234 | gzip -dc | mencoder -demuxer rawvideo -rawvideo fps=60:w=384:h=640:format=rgba -ovc x264 -o out.avi -
+
+On Ubuntu Touch:
+----------------
+    
+    mirscreencast -m /run/mir_socket --stdout --cap-interval 1 -s 384 640 | gzip -c | nc 10.42.0.209 1234
```
Comment (UniversalSuperBox) [inline userguide/advanceduse/screencasting.rst]: Please add a statement followed by two colons (as mentioned above) to explain this command and force Sphinx to create a code block.

File: userguide/advanceduse/screencasting.rst [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+
+
+References
+----------
+
+* initial source: https://wiki.ubuntu.com/Touch/ScreenRecording`
```
Comment (UniversalSuperBox) [inline userguide/advanceduse/screencasting.rst]: There's an extraneous backtick (\`) in this line

File: userguide/advanceduse/reverse-tethering.rst [hunk 6]
```diff
@@ -0,0 +1,34 @@
+Reverse tethering
+=================
+
+Some users may don't have an available wifi connection for their phone nor a data subscription from their mobile provider.
+This short tutorial will help you to connect your Ubuntu Touch to your computer to access internet.
+
+Prerequisite: Phone is connected to the computer with usb and developper mode enabled.
```
Comment (UniversalSuperBox) [inline userguide/advanceduse/reverse-tethering.rst]: Small spelling change: `developper` to `developer`

**Comment by lduboeuf** [comment]
right i think i will install sphinx locally or by using an editor to avoid wasting your time at reviewing formating text :) i'm so used to Markdown now... and github preview don't render correctly rst file i guess.

**Comment by lduboeuf** [comment]
ok seems to be good i hope

File: userguide/advanceduse/reverse-tethering.rst [hunk 7]
```diff
@@ -0,0 +1,34 @@
+Reverse tethering
+=================
+
+Some users may don't have an available wifi connection for their phone nor a data subscription from their mobile provider.
+This short tutorial will help you to connect your Ubuntu Touch to your computer to access internet.
+
+Prerequisite: Phone is connected to the computer with usb and developer mode enabled.
+
+Steps:
```
Comment (UniversalSuperBox) [inline userguide/advanceduse/reverse-tethering.rst]: Remove colons from headings.

File: userguide/advanceduse/reverse-tethering.rst [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+------
+
+1. on phone: ``android-gadget-service enable rndis``
+
+2. on computer: get your rndis ip address e.g:``hostname -I``
+
+3. on phone:
```
Comment (UniversalSuperBox) [inline userguide/advanceduse/reverse-tethering.rst]: Please capitalize the first letter of every point.

**Comment by lduboeuf** [comment]
lol, sorry :)

Other question: i would like to put an extra file ( say a bash script ) put don't know where to put it (same  folder, assets folder, other folder ?)

**Comment by UniversalSuperBox** [comment]
Hmm, good question. Looking at how [the Godot engine does it](https://github.com/godotengine/godot-docs/commit/786c89923dbc403c034cce3a0f34fb868b650135), I'd say place the file inside of a `files` directory in the same directory as the RST you're editing.

**Comment by lduboeuf** [comment]
alright, done

**Pull request merged**